digitalocean/resource_digitalocean_uptime_check.go

```diff
@@ -0,0 +1,185 @@
+package digitalocean
+
+import (
+	"context"
+	"log"
+
+	"github.com/digitalocean/godo"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+)
+
+func resourceDigitalOceanUptimeCheck() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceDigitalOceanUptimeCheckCreate,
+		ReadContext:   resourceDigitalOceanUptimeCheckRead,
+		UpdateContext: resourceDigitalOceanUptimeCheckUpdate,
+		DeleteContext: resourceDigitalOceanUptimeCheckDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:        schema.TypeString,
+				Description: "A human-friendly display name for the check.",
+				Required:    true,
+			},
+			"id": {
+				Type:     schema.TypeString,
+				Computed: true,
+				ForceNew: true,
+			},
+			"regions": {
+				Type:        schema.TypeSet,
+				Description: "An array containing the selected regions to perform healthchecks from.",
+				Optional:    true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+			"type": {
+				Type:        schema.TypeString,
+				Description: "The type of health check to perform. Enum: 'ping' 'http' 'https'",
+				Default:     "https",
+				Optional:    true,
+			},
+			"target": {
+				Type:        schema.TypeString,
+				Description: "The endpoint to perform healthchecks on.",
+				Required:    true,
+			},
+			"enabled": {
+				Type:        schema.TypeBool,
+				Default:     false,
```

Comment:
I wonder if we should default this to `true`? Seems like it would be the most common use case.

Reply:
I agree. will update- ty!